From 328097633e434caab520540c4c9a620771289a3a Mon Sep 17 00:00:00 2001
From: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Date: Thu, 18 Jan 2024 16:22:03 +0530
Subject: [PATCH] hostapd: Send unicast probe response on AP VAP with hidden
 SSID enabled.

Make probe response as unicast for 6GHz band if AP is not indicating
the actual SSID in the SSID element of its Beacon frames.

In case of MBSSID, if probed VAP is configured with hidden ssid then send probe
response as unicast.

Change-Id: Ic6bf3f3b081dfb44f2cafe6bf41a0ee297ebc21f
Signed-off-by: Rathees Kumar R Chinannan <quic_rrchinan@quicinc.com>
Signed-off-by: Sathishkumar Muruganandam <quic_murugana@quicinc.com>
---
 src/ap/beacon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -1046,7 +1046,7 @@ static void hostapd_gen_probe_resp(struc
 	 * the Probe Response frame template for the unsolicited (i.e., not as
 	 * a response to a specific request) case. */
 	if (params->req && (!is_6ghz_op_class(hapd->iconf->op_class) ||
-		    hapd->conf->ignore_broadcast_ssid))
+		    hapd_probed->conf->ignore_broadcast_ssid))
 		os_memcpy(params->resp->da, params->req->sa, ETH_ALEN);
 	else
 		os_memset(params->resp->da, 0xff, ETH_ALEN);
